axes/decorators.py

```diff
@@ -453,8 +453,12 @@ def check_request(request, login_unsuccessful):
     if failures >= FAILURE_LIMIT and LOCK_OUT_AT_FAILURE and user_lockable:
         # We log them out in case they actually managed to enter the correct
         # password
-        if hasattr(request, 'user') and request.user.is_authenticated():
-            logout(request)
+        try:
```

Comment:
I think try - catch might not be the easiest way of implementing this check. It could be implemented as a simple function that abstracts away all the logic and is easy to test with different setups:

```python
def is_authenticated(request):
    if not hasattr(request, 'user'):
        return False

    if callable(request.user.is_authenticated):
            return request.user.is_authenticated()

    return request.user.is_authenticated
```

Then it's possible to just use:

```python
if is_authenticated(request):
    logout(request)
```

Reply:
Changed the PR accordingly. FYI: my changes are on top of 1.6.1 release, is that okay ?

Reply:
Yeah it's OK. I'll merge the changes into master and make a maintenance branch for 1.6 and make a release for 1.6.2. It should be a relatively small effort :)